doc/source/tutorial/tutorial_sessions.ipyml

```diff
@@ -8,14 +8,115 @@ cells:
+
 - code: |
     from larray import *
 
 
 - markdown: |
-    ## Before To Continue
+    ## Three Kinds Of Sessions  
+
+
+- markdown: |
+    They are three ways to group objects in LArray:
     
-    If you are not yet comfortable with creating, saving and loading sessions, please read first the [Creating Sessions](tutorial_presenting_larray_objects.ipynb#Creating-Sessions) section of the tutorial before going further.  
+      * [Session](../api.rst#session): is an ordered dict-like container with special I/O methods. Although the *autocomplete*\* feature on the objects stored in the session is available in the larray-editor, it is not available in development tools like PyCharm making it cumbersome to use.
+      * [CheckedSession](../api.rst#checkedsession): provides the same methods as Session objects but are defined in a complete different way (see example below). The *autocomplete*\* feature is both available in the larray-editor and in development tools (PyCharm). In addition, the type of each stored object is protected. Optionally, it is possible to constrain the axes and dtype of arrays using ``CheckedArray``.
+      * [CheckedParameters](../api.rst#checkedparameters): is a special version of CheckedSession in which the value of all stored objects (parameters) is frozen after initialization.
+      
+     \* The *autocomplete* is the feature in which development tools try to predict the variable or function a user intends to enter after only a few characters have been typed (like word completion in cell phones).
+
+
+- markdown: |
+    ## Creating Sessions 
+
+
+- markdown: |
+    ### Session
+    
+    Create a session:
+
+
+- code: |
+    # define some scalars, axes and arrays
+    variant = 'baseline'
+    
+    country = Axis('country=Belgium,France,Germany')
+    gender = Axis('gender=Male,Female')
+    time = Axis('time=2013..2017')
+    
+    population = zeros([country, gender, time])
+    births = zeros([country, gender, time])
+    deaths = zeros([country, gender, time])
+
+
+- code: |
+    # create an empty session and objects one by one after
+    s = Session()
+    s.variant = variant
+    s.country = country
+    s.gender = gender
+    s.time = time
+    s.population = population
+    s.births = births
+    s.deaths = deaths
+    
+    print(s.summary())
+
+
+- code: |
+    # or create a session in one step by passing all objects to the constructor
+    s = Session(variant=variant, country=country, gender=gender, time=time, 
+                population=population, births=births, deaths=deaths)
+    
+    print(s.summary())
+
+
+- markdown: |
+    ### CheckedSession
+    
+    The syntax to define a checked-session is a bit specific:
+    
+    ```python
+    class MySession(CheckedSession):
+        # Variables can be declared in two ways:
+        # a) by specifying only the type of the variable (to be initialized later)
+        var1: Type
+        # b) by giving a default value
+        var2 = initialization value
```

Comment:
shouldn't you mention that the type is autodetected in that case?

Reply:
Yes, indeed